Sources/Core/Sources/Render/Mesh/FluidMeshBuilder.swift

```diff
@@ -106,9 +106,10 @@ struct FluidMeshBuilder { // TODO: Make fluid meshes look more like they do in v
     tint: SIMD3<Float>
   ) {
     let basePosition = position.relativeToChunkSection.floatVector + SIMD3<Float>(0.5, 0, 0.5)
+    let cullTop = neighbouringBlocks[.up]?.fluidId == fluid.id
 
-    // Iterate through all visible faces
-    for direction in Direction.allDirections where !cullingNeighbours.contains(direction) {
+    // Iterate through all visible faces, filtering out the top face if necessary
+    for direction in Direction.allDirections where !cullingNeighbours.filter({d in  d != .up || cullTop }).contains(direction) {
```

Comment:
The filter will get run every iteration and is an inefficient operation to use on a set. Instead, make a mutable copy of the set and remove `.up` if `!cullTop` (before the for loop). In the future I'll make this even more efficient by using a bitset (OptionSet) instead of a `Set<Direction>`, but that's not necessary for now.

Reply:
Actually, is it possible to just hook the conditional into the `case .up:` statement? Something where we just run the case if cullTop is true?

Reply:
Should be more efficient now. I'm going to look into the linux branch now.